plugins/tabletools/plugin.js

```diff
@@ -325,27 +330,34 @@
 					cell = new CKEDITOR.dom.element( mapRow[ i ] );
 
 				if ( cell.$ ) {
-					if ( cell.$.colSpan == 1 )
+					if ( cell.$.colSpan == 1 ) {
 						cell.remove();
-					// Reduce the col spans.
-					else
+					} else {
+						// Reduce the col spans.
 						cell.$.colSpan -= 1;
+					}
 
 					j += cell.$.rowSpan - 1;
 
-					if ( !row.$.cells.length )
+					if ( !row.$.cells.length ) {
 						rowsToDelete.push( row );
+					}
 				}
 			}
 		}
 
-		var firstRowCells = table.$.rows[ 0 ] && table.$.rows[ 0 ].cells;
-
 		// Where to put the cursor after columns been deleted?
 		// 1. Into next cell of the first row if any;
 		// 2. Into previous cell of the first row if any;
 		// 3. Into table's parent element;
-		var cursorPosition = new CKEDITOR.dom.element( firstRowCells[ startColIndex ] || ( startColIndex ? firstRowCells[ startColIndex - 1 ] : table.$.parentNode ) );
+		var cursorPosition;
```

Comment:
Generally `tabletools` doesn't depend on `tableselection` so we cannot assume that `tableselection` will be loaded.